### ISSUE TITLE
[Bug] Invalid lane connections by stay-in-lane

### ISSUE THREAD

**Issue by Monniasza** [CitiesSkylinesMods/TMPE#1016]
### Describe the problem
Bottom road: 3 lane 1 way from Network Extensions, upwards
Left road: tiny 2 lane street from Network Extensions, leftwards
Right and upper road: tiny 1 lane street from Network Extensions, forwards

In the screensot, lanes crisscross, despite pop-up hint promising it will not happen.

### Steps to reproduce

1. Place roads as in the description
2. Activate TM:PE → Lane Connector
3. Select intersection
4. Press CTRL+S

### Log files
Not  an exception, not required

### Savegame?
Not required

### Screenshots?
![20201014102630_1](https://user-images.githubusercontent.com/37037712/96015407-d55c6080-0e47-11eb-9c6e-cbdbd70eb63c.jpg)

### Notes or questions?
The stay in lane system should be enhanced, so any road combination works:
- If user clicks on a segment then it is set as main road
- Else largest road is set as the main road.
- Other roads are read from the furthest lane
- For each lane from far side of main road, there will be assigned a lane, counting from far side of other roads.

**Comment by krzychu124** [comment]
Yeah, something is wrong. I even changes outgoing road types to highway one lane and still the same weird result
@kianzarrin any ideas what is wrong here?

![ezgif com-gif-maker](https://user-images.githubusercontent.com/19638970/97369219-c2687800-18ac-11eb-8320-cf0bb021c317.gif)

**Comment by Monniasza** [comment]
@kianzarrin @krzychu124 When it will be fixed?

**Comment by kianzarrin** [comment]
the math is heavy and fixing this is not that important.

**Comment by aubergine10** [comment]
This won't fix problem above, however...

NExt2 roads have some weird bugs in them. You could get faster load and less lag + bugs in game by replacing NExt2 mod-based roasd with a pack of [asset-based NExt2 roads](https://steamcommunity.com/workshop/filedetails/?id=2585558081) that look and work the same. See [RON workshop page](https://steamcommunity.com/sharedfiles/filedetails/?id=2405917899) for details.
